ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -778,6 +778,12 @@ private static void verifyStreamTableJoin(final JoinInfo joinInfo, final PlanNod
           joinInfo.hasFlippedJoinCondition()
       );
 
+      if (((DataSourceNode) rightNode).isWindowed()) {
+        throw new KsqlException(
+           "Foreign-key table-table joins are not supported on windowed tables."
```

Comment:
We don't support any type of join for windowed tables, right, thus I am wondering if this is the right place for this check? (Also, we would need to check both inputs, not just the right input I guess?)

Reply:
> We don't support any type of join for windowed tables, right, thus I am wondering if this is the right place for this check?

We do support PK joins on windowed tables (push queries only).

> Also, we would need to check both inputs, not just the right input I guess?

There's a separate check in the code to ensure that you cannot join a windowed source to a non-windowed source.